.github/workflows/docker_publish.yml

```diff
@@ -0,0 +1,101 @@
+on:
+  push:
+    branches: ['gh_actions_and_ghcr_for_docker']  # todo replace with main once we know this is working
```

Comment:
Can we add the filter so that this only runs when certain files change?

Reply:
I am not sure about triggering on changes to certain files, I shall look into that

I can however get it to trigger on a GitHub Release if that helps

Reply:
I found this somewhere and need to find the link again

Reply:
https://github.community/t/is-it-possible-to-run-the-job-only-when-a-specific-file-changes/115484

Reply:
That is a nice GitHub action you found there. 

I can give this a go.

Which files would we want to monitor for changes. I guess just the source code folder and the Docker files themselves

Reply:
For this action, it's ONLY the Docker files - not the source code.  There is no reason to rebuild the images for CI (which don't include DAGMC itself) when the source code changes.   I suppose we can add a different action that adds DAGMC to one of these images on release.

Reply:
In that case it might be worth moving the Dockerfiles to a new repo as it appears they are somewhat disconnected from the DAGMC source code itself.

Reply:
But they are specifically the images we rely on for CI, so that's why it makes sense for them to stay here.  We do have the other dockerfiles repo, and that could be a place to develop more generic dockerfiles-dagmc-dependencies images.  For example, the specific hash of MOAB that we have in here right now is to advance CI and not because we would want to publish a dependency based on it.

Reply:
I've added the conditional file change trigger to the github action, thanks for finding that.

I've tried to checked it works by pushing with no changes and again pushing with superficial changes to the Dockerfiles

As you can see in the Github Actions appeared to skip both times, so I've got something a bit wrong but I think we are close